Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,402 @@
+               let startingLocation = utilityNetwork.makeElement(assetType: assetType, globalID: globalID) {
+                // Set the terminal for the location. (For our case, use the "Load" terminal.)
+                startingLocation.terminal = assetType.terminalConfiguration?.terminals.first(where: { $0.name == terminalName })
+                return startingLocation
+            } else {
+                return nil
+            }
+        }
+        
+        /// Adds a graphic to the graphics overlay.
+        /// - Parameters:
+        ///   - location: The `Point` location to place the graphic..
+        ///   - traceLocationType: The textual description of the trace location type.
+        private func addGraphic(for location: Point, traceLocationType: String) {
+            let graphic = Graphic(
+                geometry: location,
+                attributes: ["TraceLocationType": traceLocationType]
+            )
+            parametersOverlay.addGraphic(graphic)
+        }
+        
+        /// Sets the selected filter barrier category and updates the status text.
+        func selectCategory(_ category: UtilityCategory) {
+            selectedCategory = category
+            statusText = "\(category.name) selected."
+            traceEnabled = true
+        }
+        
+        /// Unselects the selected filter barrier category and updates the status text.
+        func unselectCategory(_ category: UtilityCategory) {
+            selectedCategory = nil
+            traceEnabled = hasFilterBarriers
+            statusText = "Tap on the map to add filter barriers, or run the trace directly without filter barriers."
+        }
+        
+        /// Runs a trace with the pending trace configuration and selects features in the map that
+        /// correspond to the element results.
+        @MainActor
+        func trace() async {
+            // Clear previous trace results.
+            layers.forEach { $0.clearSelection() }
+            
+            tracingActivity = .runningTrace
+            traceEnabled = false
+            
+            let configuration = makeTraceConfiguration(category: selectedCategory)
+            traceParameters.traceConfiguration = configuration
+            let traceResults: [UtilityElementTraceResult]
+            do {
+                traceResults = try await utilityNetwork
+                    .trace(using: traceParameters)
+                    .compactMap { $0 as? UtilityElementTraceResult }
+            } catch {
+                statusText = "Trace failed."
+                traceEnabled = true
+                traceCompleted = false
+                return
+            }
+            if !hasFilterBarriers, let selectedCategory {
+                statusText = "Trace with \(selectedCategory.name.lowercased()) category completed."
+            } else {
+                statusText = "Trace with filter barriers completed."
+            }
+            defer {
```

Comment:
This defer block makes the method a bit hard to read. There are 4 to 5 branches to exit the method and some of them reach this defer block, some don't.

Can you break the method down to 2 parts? One to make the configs and params, then run trace; the other to handle the trace result, and set various states?

Reply:
Done.